crates/red_knot_python_semantic/resources/mdtest/annotations/any.md

```diff
@@ -0,0 +1,68 @@
+# Any
+
+## Annotation
+
+`typing.Any` is a way to name the Any type.
+
+```py
+from typing import Any
+
+x: Any = 1
+x = "foo"
+
+def f():
+    reveal_type(x)  # revealed: Any
+```
+
+## Aliased to a different name
+
+If you alias `typing.Any` to another name, we still recognize that as a spelling of the Any type.
+
+```py
+from typing import Any as RenamedAny
+
+x: RenamedAny = 1
+x = "foo"
+
+def f():
+    reveal_type(x)  # revealed: Any
+```
+
+## Shadowed class
+
+If you define your own class named `Any`, using that in a type expression refers to your class, and
+isn't a spelling of the Any type.
+
+```py
+class Any:
+    pass
+
+x: Any
+
+def f():
+    reveal_type(x)  # revealed: Any
+
+# This verifies that we're not accidentally seeing typing.Any, since str is assignable
+# to that but not to our locally defined class.
+y: Any = "not an Any"  # error: [invalid-assignment]
+```
+
+## Subclass
+
+The spec allows you to define subclasses of `Any`, which must also resolve to the Any type.
+
+```py
+from typing import Any
+
+class Subclass(Any):
+    pass
+
+# Since Subclass is a subclass of Any, it is assignable to and from any other type, just like Any.
+x: Subclass = 1
+y: int = Subclass()
```

Comment:
I think this comment, and the line `x: Subclass = 1`, are not quite right; that latter assignment should be an error. `Subclass` is not equivalent to `Any`; it's a distinct class, which has an unknown base class. There is no "materialization" (using the term from the typing spec, to mean "replacing Any with a concrete static type") of the unknown base class that could possibly result in `1` being an instance of `Subclass`.

The second test line here is correct; since `Subclass` inherits from an unknown type, that unknown type _could_ be (could materialize to) `int` or a subclass of `int`, therefore we should allow the assignment `y: int = Subclass()`.

You can see that both [pyright](https://pyright-play.net/?enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCCKcAUCQMYA2AhgM61QDKArgEYAURcAlAFwlRBUBHVpkAHryZsoAXigBGEnCmp88lh25kgA) and [mypy](https://mypy-play.net/?mypy=latest&python=3.12&gist=63d4f40219fb0072ebd811c6ae7937db) agree with this, via their online playgrounds.

Reply:
Just for future reference, not because it's relevant to this PR: one place where I think mypy and pyright get it wrong is that they also allow the assignment `x: Fin = Sub()` in [this code sample](https://pyright-play.net/?enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCCKcANFMKgIYA2AULQMbWUDOLUAygK4BGAFETgBKAFy0oEqAlYt6AAQooajZmygAxVGMlSZ9AB4jOvKAF4oARlpwjqfOe78htAF5HNKM8af1aQA). Because `Fin` is a final class, it can have no subclasses, therefore the unknown base class of `Sub` can't materialize to `Fin`, therefore I don't think that assignment should be allowed. But this is for future: we don't even have support for final classes in red-knot at all yet.